### PR TITLE
Milestone 3: Game Logic

### PR DESCRIPTION
**Pull request by phylischumba**
## In this Milestone we:
- Created branch game_logic
- Added logic to the game and got rid of inline comments
- Created different classes for different parts of the game logic
- Introduced attr_readers in our classes

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  end
+
+  def display_board(_choice = nil, _current_player = nil)
+    puts "#{@board[0]} | #{@board[1]} | #{@board[2]}"
+    puts '---------'
+    puts "#{@board[3]} | #{@board[4]} | #{@board[5]}"
+    puts '---------'
+    puts "#{@board[6]} | #{@board[7]} | #{@board[8]} "
+  end
```
Comment (maelfosso) [inline lib/board.rb]: No `puts/gets` in classe

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def get_chip(chip)
+    chip == 'x' ? 'o' : 'x'
+  end
+
+  def board_update(_board, choice, player)
```
Comment (maelfosso) [inline lib/board.rb]: why `_board` parameter?

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Player
```
Comment (maelfosso) [inline lib/game.rb]: The file name should be identic to the class name.

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def wins?(choice)
+    @final_array.push(choice)
+    @winner = false
+    @win_arr.each do |arr|
+      @winner = true if @final_array.include?(arr[0]) && @final_array.include?(arr[1]) && @final_array.include?(arr[2])
+    end
+    @winner
+  end
```
Comment (maelfosso) [inline lib/game.rb]: Move it into the `Board` class

File: bin/main.rb [hunk 5]
```diff
@@ -1,81 +1,53 @@
...
+while game_on
+  puts "#{current_player.name} select a Position between (1-9)"
+  player_pos = gets.chomp.to_i
+  until board.position_taken?(player_pos)
+    puts 'Enter a valid position between 1-9:'
+    puts board.display_board
+    player_pos = gets.chomp.to_i
+  end
+
+  board.board_update(board, player_pos, current_player.chip)
```
Comment (maelfosso) [inline bin/main.rb]: `board` parameter is not needed

File: bin/main.rb [hunk 6]
```diff
@@ -1,81 +1,53 @@
...
+  puts board.display_board
+  if current_player.wins?(player_pos)
```
Comment (maelfosso) [inline bin/main.rb]: why not, instead of `current_player.wins?` you called something like `board.winner(player_pos)`

File: bin/main.rb [hunk 7]
```diff
@@ -1,81 +1,53 @@
...
+    game_on = false
+    puts "Congratulations#{current_player.name} You are a Tic Tac Toe Master!"
+    break
   end
-  puts "end of round #{count}"
   count += 1
+  current_player = current_player == player_one ? player_two : player_one
```
Comment (maelfosso) [inline bin/main.rb]: Move it as a method into the `Board` class
```ruby
def switch_player
   ....
end
```
and call `board.switch_player`

File: bin/main.rb [hunk 8]
```diff
@@ -1,81 +1,53 @@
...
+
+  board.board_update(board, player_pos, current_player.chip)
+  puts board.display_board
+  if current_player.wins?(player_pos)
+    game_on = false
+    puts "Congratulations#{current_player.name} You are a Tic Tac Toe Master!"
+    break
   end
-  puts "end of round #{count}"
   count += 1
```
Comment (maelfosso) [inline bin/main.rb:58]: **`[OPTIONAL]`** It will also be great to move counter into a method of `Board` class

**Review by maelfosso — CHANGES_REQUESTED** [review]
# changes required :red_circle: 

Hello @Phylis05 @imahnama 

It's working :+1: 
However, there are some issues related to the SOLID principle.
Please, read the comments below.

Kindly make the suggested changes and submit a new code review request.
Thanks

Mael FOSSO

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, Just few things to work on. 
### General
- [x] The game isn't working properly, it stops after three input


- [x] Lets write a method for when there is a draw between both players
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by whiz25 — APPROVED** [review]
## Congz team :clap:
You have made all the requested changes and your game works as required by the requirements. It's a :+1: from me. Merge and proceed to the next milestone.

### Martin Afani

**Pull request merged**